src/server/eventsupplier.cpp

```diff
@@ -213,9 +217,11 @@ SendEventType EventSupplier::detect_and_push_events(DeviceImpl *device_impl,stru
 	{
 		switch (*ite)
 		{
-			case 5:
-			if (archive5_subscription >= EVENT_RESUBSCRIBE_PERIOD)
-				attr.remove_client_lib(5,string(EventName[ARCHIVE_EVENT]));
+            //TODO extract class hierarchy based on version!!!
+            case 6:
+            case 5:
+                if (archive5_subscription >= EVENT_RESUBSCRIBE_PERIOD)
+                    attr.remove_client_lib(5, string(EventName[ARCHIVE_EVENT]));
```

Comment:
I think we might encounter issues when *ite == 6 here...
I would replace this line with:
`attr.remove_client_lib(*ite, string(EventName[ARCHIVE_EVENT]));`